app/views/play/level/PlayGameDevLevelView.coffee

```diff
@@ -48,17 +49,22 @@ module.exports = class PlayGameDevLevelView extends RootView
     @supermodel.on 'update-progress', (progress) =>
       @state.set({progress: (progress*100).toFixed(1)+'%'})
     @level = new Level()
-    @session = new LevelSession()
+    @session = new LevelSession({ _id: @sessionID })
     @gameUIState = new GameUIState()
     @courseID = utils.getQueryVariable 'course'
     @courseInstanceID = utils.getQueryVariable 'course-instance'
     @god = new God({ @gameUIState, indefiniteLength: true })
-    @levelLoader = new LevelLoader({ @supermodel, @levelID, @sessionID, observing: true, team: TEAM, @courseID })
-    @supermodel.setMaxProgress 1 # Hack, why are we setting this to 0.2 in LevelLoader?
-    @listenTo @state, 'change', _.debounce @renderAllButCanvas
-    @updateDb = _.throttle(@updateDb, 1000)
 
-    @levelLoader.loadWorldNecessities()
+    new Promise((accept,reject) => @session.fetch().then(accept, reject)).then (sessionData) =>
+      api.levels.getByOriginal(sessionData.level.original)
+    .then (levelData) =>
+      @levelID = levelData.slug
+      @levelLoader = new LevelLoader({ @supermodel, @levelID, @sessionID, observing: true, team: TEAM, @courseID })
```

Comment:
Add `@session` to `@supermodel` with `@supermodel.registerModel` before here, and then the LevelLoader should grab the loaded session so it doesn't load twice.